json-path/src/main/java/com/jayway/jsonpath/Predicate.java

```diff
@@ -22,6 +25,7 @@
 public interface Predicate {
 
     boolean apply(PredicateContext ctx);
+    void getRelationalExprValues(ArrayList<SimpleEntry<String,String>> valuesMap);
```

Comment:
Change all interfaces to take and return List<SimpleEntry<String, String>> - the concrete implement, the one creating the list, will be the one that should decide to use a ArrayList

Reply:
Done